File: DevExpress-Demo/VB/PropertyGridDemo.Wpf/Helpers/GridColumnHelper.vb
Imports System.Windows
Imports System.Windows.Controls
Imports DevExpress.Mvvm.UI.Interactivity

Namespace PropertyGridDemo

    Public Class GridColumnHelper
        Inherits Behavior(Of Grid)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.SizeChanged, AddressOf AssociatedObject_SizeChanged
        End Sub

        Private Sub AssociatedObject_SizeChanged(ByVal sender As Object, ByVal e As SizeChangedEventArgs)
            AssociatedObject.ColumnDefinitions(1).Width = New GridLength(e.NewSize.Width / 3, GridUnitType.Pixel)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/CS/PropertyGridDemo.Wpf/Helpers/GridColumnHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using DevExpress.Mvvm.UI.Interactivity;

namespace PropertyGridDemo
{
    public class GridColumnHelper : Behavior<Grid> {
        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.SizeChanged += AssociatedObject_SizeChanged;
        }

        void AssociatedObject_SizeChanged(object sender, System.Windows.SizeChangedEventArgs e) { AssociatedObject.ColumnDefinitions[1].Width = new GridLength(e.NewSize.Width / 3, GridUnitType.Pixel); }
    }
}
